### ISSUE TITLE
pFocusTrap always enabled in Drawer

### ISSUE THREAD

**Issue by ddpdoj** [primefaces/primeng#19422]
### Describe the bug

Currently, `pFocusTrap` is always enabled for instances of `p-drawer`, which for most cases is desirable. However, for the case when `[dismissible]="false" [modal]="false"`,  moving focus from the non-dismissible, non-modal drawer become impossible. This edge case is overcome in similar components by adding an input variable that allows `pFocusTrapDisabled` to be set, i.e. the [dialog component](https://github.com/primefaces/primeng/blob/master/packages/primeng/src/dialog/dialog.ts#L78C26-L78C44).

### Pull Request Link

https://github.com/primefaces/primeng/pull/19424

### Reason for not contributing a PR

- [ ] Lack of time
- [ ] Unsure how to implement the fix/feature
- [ ] Difficulty understanding the codebase
- [ ] Other

### Reproducer

https://stackblitz.com/edit/tztfnue8-rxgksn9k?file=src%2Fapp%2Fdrawer-template-demo.html

### Environment

MacOS using Angular v21 and PrimeNg v21

### Angular version

21.1.5

### PrimeNG version

v21

### Node version

24.10.0

### Browser(s)

Chrome

### Steps to reproduce the behavior

1. Set `dismissible` and `modal` to be `false` on a `p-drawer` component
2. Tab through focusable elements on the page
3. Notice how focusTrap is working as expected, which is not desirable in this case

### Expected behavior

Addition of an input variable to `p-drawer` that allows for the disabling of the focusTrap.
